core/src/OpenMPTarget/Kokkos_OpenMPTarget_Instance.cpp

```diff
@@ -120,6 +122,10 @@ void OpenMPTargetInternal::impl_initialize() {
   omp_set_num_teams(512);
 #endif
 #endif
+
+  using Kokkos::Impl::get_gpu;
+  const int device_num = get_gpu(settings);
+  omp_set_default_device(device_num);
```

Comment:
Move this to ` OpenMPTarget::impl_initialize`.  We only want to run it once in the program.

Reply:
This is already in `impl_initialize` right? or do you mean somewhere elase?

Reply:
This is `OpenMPTargetInternal::impl_initialize` not `OpenMPTarget::impl_initialize`. Currently, the latter calls the former for the singleton but every other execution space instance would only call the former. Setting the device should be done by the latter, though, so it's only done once by the singleton.
Hence, it's better to move this call to `OpenMPTarget::impl_initialize`.

Reply:
Thanks @masterleinad for the explanation. 
I moved the routine to `OpenMPTarget::impl_initialize`